File: server/models/ContactSubmission.js
const mongoose = require('mongoose');

const ContactSubmissionSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  email: {
    type: String,
    required: true,
    lowercase: true,
    trim: true
  },
  phone: {
    type: String,
    trim: true,
    default: ''
  },
  company: {
    type: String,
    trim: true,
    default: ''
  },
  subject: {
    type: String,
    required: true,
    trim: true
  },
  message: {
    type: String,
    required: true,
    trim: true
  },
  status: {
    type: String,
    enum: ['NEW', 'IN_PROGRESS', 'RESOLVED'],
    default: 'NEW'
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('ContactSubmission', ContactSubmissionSchema);

File: server-java/src/main/java/com/denesens/model/ContactSubmission.java
package com.denesens.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Document(collection = "contactsubmissions")
public class ContactSubmission {
    @Id
    @JsonProperty("_id")
    private String id;
    
    private String name;
    private String email;
    private String phone = "";
    private String company = "";
    private String subject;
    private String message;
    private String status = "NEW"; // NEW, IN_PROGRESS, RESOLVED

    @CreatedDate
    private Instant createdAt;
}
